src/Sulu/Bundle/SearchBundle/EventListener/ReindexListener.php

```diff
@@ -0,0 +1,171 @@
+<?php
+/*
+ * This file is part of the Sulu CMS.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\SearchBundle\EventListener;
+
+use Massive\Bundle\SearchBundle\Search\Event\HitEvent;
+use Sulu\Component\Webspace\Analyzer\RequestAnalyzerInterface;
+use Sulu\Component\Util\SuluNodeHelper;
+use Sulu\Component\PHPCR\SessionManager\SessionManagerInterface;
+use Sulu\Component\Content\Mapper\ContentMapperInterface;
+use Massive\Bundle\SearchBundle\Search\SearchManagerInterface;
+use Sulu\Component\Webspace\Manager\WebspaceManagerInterface;
+use Massive\Bundle\SearchBundle\Search\Event\IndexRebuildEvent;
+use Sulu\Component\Content\StructureManagerInterface;
+use Sulu\Component\Content\Structure;
+
+/**
+ * Listen to for new hits. If document instance of structure
+ * prefix the current resource locator prefix to the URL
+ */
+class ReindexListener
+{
+    /**
+     * @var SessionManagerInterface
+     */
+    private $sessionManager;
+
+    /**
+     * @var ContentMapperInterface
+     */
+    private $contentMapper;
+
+    /**
+     * @var SearchManagerInterface
+     */
+    private $searchManager;
+
+    /**
+     * @var WebspaceManagerInterface
+     */
+    private $webspaceManager;
+
+    /**
+     * @var StructureManagerInterface
+     */
+    private $structureManager;
+
+    /**
+     * @var SuluNodeHelper
+     */
+    private $nodeHelper;
+
+    /**
+     * @var string
+     */
+    private $pageIndexName;
+
+    /**
+     * @var string
+     */
+    private $snippetIndexName;
+
+    public function __construct(
+        SessionManagerInterface $sessionManager,
+        ContentMapperInterface $contentMapper,
+        SearchManagerInterface $searchManager,
+        WebspaceManagerInterface $webspaceManager,
+        StructureManagerInterface $structureManager,
+        SuluNodeHelper $nodeHelper,
+        $pageIndexName,
+        $snippetIndexName
+    )
+    {
+        $this->sessionManager = $sessionManager;
+        $this->contentMapper = $contentMapper;
+        $this->searchManager = $searchManager;
+        $this->webspaceManager = $webspaceManager;
+        $this->structureManager = $structureManager;
+        $this->nodeHelper = $nodeHelper;
+        $this->pageIndexName = $pageIndexName;
+        $this->snippetIndexName = $snippetIndexName;
+    }
+
+    /**
+     * Prefix url of document with current resourcelocator prefix
+     * @param HitEvent $event
+     */
+    public function onIndexRebuild(IndexRebuildEvent $event)
+    {
+        $output = $event->getOutput();
+        $purge = $event->getPurge();
+        $filter = $event->getFilter();
+        $session = $this->sessionManager->getSession();
+
+        $sql2 = 'SELECT * FROM [nt:unstructured] AS a WHERE [jcr:mixinTypes] = "sulu:page" OR [jcr:mixinTypes] = "sulu:snippet"';
```

Comment:
Is it a good idea to have this logic in the SearchBundle? Shouldn't that be split into parts for the content and snippet bundle? This way e.g. the index for Contacts and so on are not rebuild.